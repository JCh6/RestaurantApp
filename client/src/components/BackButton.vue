<template>
    <v-row class="my-3" :justify="justify">
        <v-btn :color="color" elevation="2" @click="$router.push(route)">
            <v-icon class="mr-2">{{ icon }}</v-icon>
            {{ msg }}
        </v-btn>
    </v-row>
</template>

<script>
export default {
    name: "BackButton",
    props: {
        icon: String,
        msg: String,
        justify: String,
        route: String,
        color: String,
    },
};
</script>