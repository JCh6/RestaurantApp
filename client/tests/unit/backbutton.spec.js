import Vue from 'vue';
import Vuetify from 'vuetify';
import BackButton from '@/components/BackButton.vue';
import { createLocalVue, mount } from '@vue/test-utils';

Vue.use(Vuetify);

describe('BackButton.vue', () => {
    const localVue = createLocalVue()
    let vuetify;

    beforeEach(() => {
        vuetify = new Vuetify()
    })

    const mountFunction = options => {
        return mount(BackButton, {
            localVue,
            vuetify,
            ...options,
        })
    }

    it('should have a custom msg, icon, route, color and match snapshot', () => {
        const wrapper = mountFunction({
            propsData: {
                msg: "Back to Calendar",
                icon: "Calendar",
                color: "primary",
            },
        })

        expect(wrapper.html()).toMatchSnapshot();
    })
})