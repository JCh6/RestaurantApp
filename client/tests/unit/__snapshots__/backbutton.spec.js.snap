// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton.vue should have a custom msg, icon, route, color and match snapshot 1`] = `
<div class="row my-3"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light elevation-2 v-size--default primary"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mr-2 material-icons theme--light">Calendar</i>
        Back to Calendar
    </span></button></div>
`;
